include/heuristic/HeuristicMapper.hpp

```diff
@@ -13,18 +13,45 @@ class HeuristicMapper: public Mapper {
 public:
     using Mapper::Mapper; // import constructors from parent class
 
+    /**
+     * @brief map the circuit passed at initialization to the architecture
+     *
+     * @param config the settings for this mapping run (controls e.g. layering methods, pre- and post-optimizations, etc.)
+     */
     void map(const Configuration& config) override;
 
+    /**
+     * @brief struct representing one node in the A* search containing info about swaps, mappings and costs
+     */
     struct Node {
-        unsigned long                        costFixed        = 0;
-        double                               costHeur         = 0.;
-        double                               lookaheadPenalty = 0.;
-        double                               costTotal        = 0.;
-        std::array<short, MAX_DEVICE_QUBITS> qubits{};    // get qubit at specific location
+        /** cost of all swaps in the node */
+        unsigned long costFixed = 0;
+        /** heuristic cost expected for future swaps needed in current circuit layer */
+        double costHeur = 0.;
+        /** heuristic cost expected for future swaps needed in later circuit layers (further layers contribute less) */
+        double lookaheadPenalty = 0.;
+        double costTotal        = 0.;
+        /**
+         * containing the logical qubit currently mapped to each physical qubit.
+         * `qubits[physical_qubit] = logical_qubit`
+         *
+         * The inverse of `locations`
+         */
+        std::array<short, MAX_DEVICE_QUBITS> qubits{}; // get qubit at specific location
```

Comment:
```suggestion
        std::array<short, MAX_DEVICE_QUBITS> qubits{};
```
I'd argue that these comments are superfluous now.